Character

CJPoll opened the pull request:
First stab at creating a character entity.

Discussion welcome.

thebriz24 committed:
Add ability to add parameters to tests

thebriz24 committed:
Delete vcs.xml

CJPoll committed:
Add a script for running the test suites

CJPoll committed:
Remove parameterized tests

The ParameterizedTest stuff was dependent on the script being run
manually, being set up in the __main__ of the script. This isn't going
to be the case with using the `discover` test runner.

CJPoll committed:
Actually test stuff when running tests

I forgot to have the test inherit from unittest.TestCase after removing
the parameterized test stuff. Whoops.

CJPoll committed:
First stab at character entity

CJPoll committed:
Readability improvement

Instead of calling the amount `xp` when (in/de)creasing XP, call the
param `amount` since that's what it actually is

CJPoll committed:
Fix copy/pasted test name _derp_

CJPoll committed:
Fix broken test from fixing broken test name

thebriz24 commented on src/character.py:
```diff
@@ -0,0 +1,64 @@
+    def buffStat(self, stat, amount):
+        self.statBuffs[stat] += amount
+
+    def buffsOn(self, stat):
```
I'm not sure what I want this name to be. I'd say statBuffs but that'd be a bit confusing with buffStats above this.

CJPoll replied:
_exactly_ what I thought

CJPoll replied:
`buffsOn("body")` is the current form - I'm open to suggestions.

thebriz24 reviewed (commented):
Maybe a little something. If we don't feel like changing it that's fine too and I'll approve it.

thebriz24 commented:
Maybe whichBuffOn("body")? Or quantity?

CJPoll committed:
Improve naming of poorly named method

CJPoll committed:
Improve naming of things

More consistent function names (increaseXXX, decreaseXXX, currentXXX)
Private nstance variables use Python conventions of `_varname`

CJPoll committed:
Update test names to reflect new method names

thebriz24 reviewed (approved):
Solid
